### ISSUE TITLE
Weapon damaged during a missed attack

### ISSUE THREAD

**Issue by BaumgeistOne** [CleverRaven/Cataclysm-DDA#53527]
### Describe the bug

"You swing wildly and miss" and "The weapon is damaged by the force of the blow" contradict each other. If you dont hit anything, there is no "force of the blow".

### Steps to reproduce

1. Get a melee weapon
2. Get into some fights
3. Get "lucky"

### Expected behavior

Weapons should only ever be damaged when they hit something.

### Versions and configuration

- OS: Linux
    - OS Version: Distributor ID: Ubuntu; Description: Ubuntu 20.04.3 LTS; Release: 20.04; Codename: focal; 
- Game Version: 0.F-4363-g8b7a449e97 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Bionic Professions [package_bionic_professions],
    Blaze Industries [blazeindustries],
    C.R.I.T Expansion Mod [crt_expansion],
    No Hope [no_hope],
    Extra Mutated Scenarios [extra_mut_scens],
    No Rail Stations [No_Rail_Stations],
    No Fungal Growth [no_fungal_growth],
    Bionic Slots [cbm_slots],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills]
]

**Comment by GoLoT** [comment]
I think the last time this was brought up it was said that this is intended as the weapon could have hit something else (the ground) or simply was under strain from your swing. Anyway, the fix should be as easy as adding an extra check to:

https://github.com/CleverRaven/Cataclysm-DDA/blob/086dcf87a5fb677269846b7ed446e1149486424d/src/melee.cpp#L845-L849

It should be `if( hits && !t.is_hallucination() )`. Or the if block (L846-849) can be moved inside the previous else block (after L839).

**Comment by Photoloss** [comment]
> intended as the weapon could have hit something else (the ground) or simply was under strain from your swing.

That only makes sense for FLIMSY or very long weapons. A knife won't break from strain, heck a shard of glass or porcelain doesn't. Said shard slipping out of your hand or you stumbling and falling represent more critical failures and need to be handled separately.

**Comment by NetSysFire** [comment]
![image](https://user-images.githubusercontent.com/59517351/146793597-3c5951cc-4bc9-4d0c-86b2-2c796b7814ee.png)

Also happens to strikes that actually do not happen. (Also note the missing terminal punctuation)

**Comment by Demivan** [comment]
I have created a merge request to fix this: #54439. Didn't even see this issue :)

**Comment by Demivan** [comment]
I agree with Photoloss that additional failures (hitting a wall or failed flimsy weapon) should be handled separately.

**Comment by BaumgeistOne** [comment]
**[kevingranade](https://github.com/kevingranade)** commented in https://github.com/CleverRaven/Cataclysm-DDA/pull/54439#issuecomment-1013739532

> A miss does not mean no contact was made, it means no damage was dealt. There's room for some more nuance here, but "your weapon is only damaged when used properly" is worse than indiscriminately applying damage.

Well, "missing" for most people indicates that no contact was made.

Thus one suggestion to fix my issue would be to change the "You miss"-message to something like:
"You hit ineffectively."
Then the following "The weapon is damaged by the force of the blow." message would make sense. :)

One more interesting question then would be if "dodging" an attack means completely avoiding contact, or just rendering the attack ineffective.

**Comment by GuardianDll** [comment]
Maybe add the `you hit the *terrain_where_the_enemy_stand*` text instead?

**Comment by RenechCDDA** [comment]
Just needs some additional messages for these cases. (Probably pass an argument to the item damage function stating whether it was a miss or hit or whatever so the message can make sense)
